src/core/common/code_utils.hpp

```diff
@@ -44,41 +44,75 @@
 #define otDEFINE_ALIGNED_VAR(name, size, align_type)            \
     align_type name[(((size) + (sizeof (align_type) - 1)) / sizeof (align_type))]
 
-/* Check the input `ERR` against zero and go to `exit` label if it is non-zero. */
-#define SuccessOrExit(ERR)                      \
-  do {                                          \
-    if ((ERR) != 0) {                           \
-      goto exit;                                \
-    }                                           \
-  } while (false)
+/**
+ *  This checks for the specified status, which is expected to
+ *  commonly be successful, and branches to the local label 'exit' if
+ *  the status is unsuccessful.
+ *
+ *  @param[in]  aStatus     A scalar status to be evaluated against zero (0).
+ *
+ */
+#define SuccessOrExit(aStatus)                      \
+    do                                              \
```

Comment:
We can consider adding an action to `SuccessOrExit()` as well. We don't have an immediate use for it, but I think it can be helpful... It also makes it similar to `VerifyOrExit()`. Thoughts?